### PR TITLE
use converters for AKS SDK agentpool definition

### PR DESCRIPTION
**Pull request by zmalik**
**What type of PR is this?**

/kind cleanup

**What this PR does / why we need it**:
Azure SDK provides two ways to manage agentpool spec for an AKS cluster
- containerservice.ManagedClusterAgentPoolProfileProperties
- containerservice.ManagedClusterAgentPoolProfile

`ManagedClusterAgentPoolProfile` is used for ManagedCluster reconcile https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/managedclusters/managedclusters.go#L267 

And `ManagedClusterAgentPoolProfileProperties` is used in AgentPools reconcile https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/agentpools/agentpools.go#L77

we generally want to favor AgentPools reconcile but the first-time creation of pools might fell into the  ManagedCluster reconcile if it's triggered first. This forces us to update all `Immutable` properties in `ManagedCluster` reconcile otherwise the agentpool might end up in an error state. 

As `containerservice.ManagedClusterAgentPoolProfileProperties` and `containerservice.ManagedClusterAgentPoolProfile` are two different types we cannot have one method and re-use it. We already unified the `AgentPoolSpec` generation in https://github.com/kubernetes-sigs/cluster-api-provider-azure/pull/1976

**Which issue(s) this PR fixes**:
This PR proposes to use a converters package helper method to gather at one place how AgentPool Spec for Azure SDK is generated. This will make it easier to do changes in the future where contributors might miss one of https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/agentpools/agentpools.go or 
https://github.com/kubernetes-sigs/cluster-api-provider-azure/blob/main/azure/services/managedclusters/managedclusters.go
**Special notes for your reviewer**:

**TODOs**:

- [ ] squashed commits
- [ ] includes documentation
- [ ] adds unit tests

**Release note**:

```release-note
NONE
```

### CODE REVIEW

File: azure/converters/managedagentpool.go [hunk 1]
```diff
@@ -0,0 +1,69 @@
+/*
+Copyright 2019 The Kubernetes Authors.
```
Comment (CecileRobertMichon) [inline azure/converters/managedagentpool.go:2]:
```suggestion
Copyright 2022 The Kubernetes Authors.
```

File: azure/converters/managedagentpool_test.go [hunk 2]
```diff
@@ -0,0 +1,154 @@
+/*
+Copyright 2021 The Kubernetes Authors.
```
Comment (CecileRobertMichon) [inline azure/converters/managedagentpool_test.go:2]:
```suggestion
Copyright 2022 The Kubernetes Authors.
```

File: azure/converters/managedagentpool_test.go [hunk 3]
```diff
@@ -0,0 +1,154 @@
...
+		expect func(*GomegaWithT, containerservice.ManagedClusterAgentPoolProfile)
+	}{
+		{
+			name: "Should set all values correctly",
+			pool: azure.AgentPoolSpec{
+				SKU:               "Standard_D2s_v3",
+				OSDiskSizeGB:      100,
+				Replicas:          2,
+				Version:           to.StringPtr("1.22.6"),
+				VnetSubnetID:      "/subscriptions/00000000-0000-0000-0000-000000000000/resourceGroups//providers/Microsoft.Network/virtualNetworks//subnets/",
```
Comment (Jont828) [inline azure/converters/managedagentpool_test.go:41]: Nit: should there be double slashes for `/resourceGroups//providers` and `virtualNetworks//subnets/"`?

Comment (zmalik) [reply azure/converters/managedagentpool_test.go:41]: there should not be, I will fix it. Thanks

**Review by CecileRobertMichon — COMMENTED** [review]
lgtm! thanks for doing this, it will help me make the code cleaner in #2168 

/assign @Jont828 

please squash commits!

**Comment by CecileRobertMichon** [comment]
@zmalik AKS e2e test failed, not sure if it was a flake or related to a change in the PR, PTAL https://prow.k8s.io/view/gs/kubernetes-jenkins/pr-logs/pull/kubernetes-sigs_cluster-api-provider-azure/2167/pull-cluster-api-provider-azure-e2e-exp/1503662573193531392

**Comment by zmalik** [comment]
I will take a look, could be related as we are touching the code for first-time cluster creation also.

**Comment by zmalik** [comment]
tests look fine now, will run it again just in case.

/test pull-cluster-api-provider-azure-e2e-exp

**Pull request merged**